src/main/java/com/teragrep/pth_06/planner/walker/conditions/IndexStatementCondition.java

```diff
@@ -97,9 +97,9 @@ public Condition condition() {
 
             for (final Table<?> table : tableSet) {
                 // create category temp table for this pattern match table
-                final CategoryTable categoryTable = new Created(
-                        new WithFilterTypes(new CategoryTableImpl(config, table, tokenizedValue))
```

Comment:
Now that I think about it, perhaps the older way was better to reduce temporal coupling?

Reply:
yes, just with better naming.